src/plugin/file/zip/ui/zipprocessstep.js

```diff
@@ -0,0 +1,200 @@
+goog.provide('plugin.file.zip.ui.ZIPProcessStep');
+goog.provide('plugin.file.zip.ui.ZIPProcessStepCtrl');
+
+goog.require('goog.log');
+goog.require('os.alert.AlertEventSeverity');
+goog.require('os.alert.AlertManager');
+goog.require('os.defines');
+goog.require('os.file.File');
+goog.require('os.ui.Module');
+goog.require('os.ui.file.method.UrlMethod');
+goog.require('os.ui.wiz.step.AbstractWizardStep');
+goog.require('os.ui.wiz.step.WizardStepEvent');
+goog.require('os.ui.wiz.wizardPreviewDirective');
+goog.require('plugin.file.zip');
+goog.require('plugin.file.zip.ZIPParserConfig');
+
+
+/**
+ * ZIP import file selection step
+ *
+ * @extends {os.ui.wiz.step.AbstractWizardStep.<plugin.file.zip.ZIPParserConfig>}
+ * @constructor
+ */
+plugin.file.zip.ui.ZIPProcessStep = function() {
+  plugin.file.zip.ui.ZIPProcessStep.base(this, 'constructor');
+  this.template = '<zipprocessstep></zipprocessstep>';
+  this.title = 'Unzip file(s)';
+};
+
+
+goog.inherits(plugin.file.zip.ui.ZIPProcessStep, os.ui.wiz.step.AbstractWizardStep);
+
+
+/**
+ * @inheritDoc
+ */
+plugin.file.zip.ui.ZIPProcessStep.prototype.finalize = function(config) { };
+
+
+/**
+ * The ZIP import file selection step directive
+ *
+ * @return {angular.Directive}
+ */
+plugin.file.zip.ui.processStepDirective = function() {
+  return {
+    restrict: 'E',
+    replace: true,
+    templateUrl: os.ROOT + 'views/plugin/zip/zipprocessstep.html',
+    controller: plugin.file.zip.ui.ZIPProcessStepCtrl,
+    controllerAs: 'processStep'
+  };
+};
+
+
+/**
+ * Add the directive to the module
+ */
+os.ui.Module.directive('zipprocessstep', [plugin.file.zip.ui.processStepDirective]);
+
+
+
+/**
+ * Controller for the ZIP import file selection step
+ *
+ * @param {!angular.Scope} $scope
+ * @param {!angular.JQLite} $element
+ * @param {!angular.$timeout} $timeout
+ * @constructor
+ * @ngInject
+ */
+plugin.file.zip.ui.ZIPProcessStepCtrl = function($scope, $element, $timeout) {
```

Comment:
I'm not sure it's necessary to have separate steps for reading the ZIP and allowing the user to pick files. The file picker can tell them how many files are present in the ZIP, or if something bad happened, and reduces the import process by an extra click. I realize that doesn't seem like much, but keeping workflows simple where reasonable should always be a priority.

My suggestion would be to show a "Loading ZIP..." message with a spinner while the file loads, then switch to the checklist UI when loading completes.

Reply:
Removed!